docs/learn/expert-guides/universal-profile/extend-profile-functionalities.md

```diff
@@ -0,0 +1,555 @@
+
+### Step 1: Create a Contract Supporting a Specific InterfaceId
+
+First, you need to create a Solidity contract that implements the supportsInterface function. This function should return true for at least one specific interfaceId, in this case, a dummy interfaceId like `0xaabbccdd`.
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract InterfaceIdExtension {
+    bytes4 constant DUMMY_INTERFACE_ID = 0xaabbccdd;
+
+    function supportsInterface(bytes4 interfaceId) external pure returns (bool) {
+        return interfaceId == DUMMY_INTERFACE_ID;
+    }
+
+}
+```
+
+The contract can be compiled with any blockchain development environment like Hardhat or Foundry and extract the abi and bytecode. Create `SupportsInterface.json` in the same directory of the main file.
+
+<details>
+    <summary>The ABI and the bytecode of the contract: </summary>
+
+```json
+{
+  "abi": [
+    {
+      "inputs": [
+        {
+          "internalType": "bytes4",
+          "name": "interfaceId",
+          "type": "bytes4"
+        }
+      ],
+      "name": "supportsInterface",
+      "outputs": [
+        {
+          "internalType": "bool",
+          "name": "",
+          "type": "bool"
+        }
+      ],
+      "stateMutability": "pure",
+      "type": "function"
+    }
+  ],
+  "bytecode": "0x608060405234801561001057600080fd5b50610110806100206000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c806301ffc9a714602d575b600080fd5b607f60383660046093565b7fffffffff00000000000000000000000000000000000000000000000000000000167faabbccdd000000000000000000000000000000000000000000000000000000001490565b604051901515815260200160405180910390f35b60006020828403121560a457600080fd5b81357fffffffff000000000000000000000000000000000000000000000000000000008116811460d357600080fd5b939250505056fea2646970667358221220f5bcd4c144943e9ab139d02d2609c021af3361f19e8d47a551246e3678d74e9264736f6c63430008110033"
+}
+```
+
+</details>
+
+### Step 2: Deploy the Extension Contract
+
+We will use the abi and bytecode to deploy the extension contract
+
+```js
+import { abi as SupportsInterfaceABI } from './SupportsInterface.json';
+import { bytecode as SupportsInterfaceBytecode } from './SupportsInterface.json';
+
+const supportsInterfaceFactory = new ethers.ContractFactory(
+  SupportsInterfaceABI,
+  SupportsInterfaceBytecode,
+  signer,
+);
+
+const supportsInterfaceContract = await supportsInterfaceFactory.deploy();
+await supportsInterfaceContract.deployTransaction.wait();
+
+const supportsInterfaceExtensionAddress = supportsInterfaceContract.addres;
+```
+
+### Step 3: Encode Function Selector and Store in Universal Profile
+
+:::info ments
+
+The address calling the `setData(..)` function needs to have `ADDEXTENSIONS` and `CHANGEEXTENSIONS` permission otherwise the call will fail. Check the [keyManager guides](../key-manager/get-controller-permissions.md) to learn more about permissions.
```

Comment:
```suggestion
The address calling the `setData(..)` function needs to have `ADDEXTENSIONS` and `CHANGEEXTENSIONS` permissions. Otherwise the call will fail. Check the [keyManager guides](../key-manager/get-controller-permissions.md) to learn more about permissions.
```